src/optimizer/filter_combiner.cpp

```diff
@@ -590,6 +690,23 @@ FilterResult FilterCombiner::AddTransitiveFilters(BoundComparisonExpression &com
 	// get the LHS and RHS nodes
 	Expression *left_node = GetNode(comparison.left.get());
 	Expression *right_node = GetNode(comparison.right.get());
+    if (right_node->type == ExpressionType::OPERATOR_CAST) {
```

Comment:
Which exact case are you trying to detect here? This seems very specific. Could this be generalized?